tests/test_sensor.py

```diff
@@ -0,0 +1,152 @@
+# License: MIT
+# Copyright © 2025 Frequenz Energy-as-a-Service GmbH
+
+"""Tests for the Sensor and sensor data classes."""
+
+from datetime import datetime, timedelta, timezone
+
+import pytest
+
+from frequenz.client.microgrid import Lifetime, SensorId
+from frequenz.client.microgrid.sensor import Sensor
+
+
+@pytest.fixture
+def now() -> datetime:
+    """Get the current time."""
+    return datetime.now(timezone.utc)
+
+
+def test_sensor_creation_defaults() -> None:
+    """Test Sensor defaults are as expected."""
+    sensor = Sensor(id=SensorId(1))
+
+    assert sensor.id == SensorId(1)
+    assert sensor.name is None
+    assert sensor.manufacturer is None
+    assert sensor.model_name is None
+    assert sensor.operational_lifetime == Lifetime()
+
+
+def test_sensor_creation_full(now: datetime) -> None:
+    """Test Sensor creation with all fields."""
+    start = now
+    end = start + timedelta(days=1)
+    sensor = Sensor(
+        id=SensorId(1),
+        name="test-sensor",
+        manufacturer="Test Manufacturer",
+        model_name="Test Model",
+        operational_lifetime=Lifetime(
+            start=start,
+            end=end,
+        ),
+    )
+
+    assert sensor.id == SensorId(1)
+    assert sensor.name == "test-sensor"
+    assert sensor.manufacturer == "Test Manufacturer"
+    assert sensor.model_name == "Test Model"
+    assert sensor.operational_lifetime.start == start
+    assert sensor.operational_lifetime.end == end
+
+
+@pytest.mark.parametrize(
+    "name,expected_str",
+    [(None, "<Sensor:SID1>"), ("test-sensor", "<Sensor:SID1:test-sensor>")],
+    ids=["no-name", "with-name"],
+)
+def test_sensor_str(name: str | None, expected_str: str) -> None:
+    """Test string representation of a sensor."""
+    sensor = Sensor(
+        id=SensorId(1),
+        name=name,
+        manufacturer="Test Manufacturer",
+        model_name="Test Model",
+        operational_lifetime=Lifetime(
+            start=datetime.now(timezone.utc),
+            end=datetime.now(timezone.utc) + timedelta(days=1),
+        ),
+    )
+    assert str(sensor) == expected_str
+
+
+_SENSOR = Sensor(
+    id=SensorId(1),
+    name="test",
+    manufacturer="Test Mfg",
+    model_name="Model A",
+)
+
+_DIFFERENT_NAME = Sensor(
+    id=_SENSOR.id,
+    name="different",
+    manufacturer=_SENSOR.manufacturer,
+    model_name=_SENSOR.model_name,
+)
+
+_DIFFERENT_ID = Sensor(
+    id=SensorId(2),
+    name=_SENSOR.name,
+    manufacturer=_SENSOR.manufacturer,
+    model_name=_SENSOR.model_name,
+)
+
+_DIFFERENT_BOTH_ID = Sensor(
+    id=SensorId(2),
+    name=_SENSOR.name,
+    manufacturer=_SENSOR.manufacturer,
+    model_name=_SENSOR.model_name,
+)
```

Comment:
is there a typo here?  They both seem to be the same.

Reply:
Ah, good catch. No, in the first version I included the microgrid ID too, as in 0.17 it is included, but to get it I needed to do an extra call to `GetMicrogridMetadata` just to get the microgrid ID. Since adding the microgrid ID in the next version is a backwards-compatible change I opted to leave the microgrid ID out, but forgot to remove the case for "both ID". Will remove.